src/Power_Up_Sequence.md

```diff
@@ -1,62 +1,397 @@
-
-Cart RAM the first time it is accessed on a real Game Boy contains random data.
-It will only contain known data if the Game Boy code initializes it to some value.
+When the Game Boy is powered up, the CPU starts executing instuctions at address $0000—**not** $0100!
+A program called the *boot ROM*, burned inside the CPU, is mapped "over" the cartridge ROM at first, which is responsible for the boot-up animation played before control is handed over to the cartridge's ROM.
 
+There exist 8 different known official boot ROMs:
+
+Name | Size (bytes) | Notes
+-----|--------------|------------------------------------------------------------
+DMG0 | 256          | Blinks on failed checks, no ®
+DMG  | 256          | |
+MGB  | 256          | One-byte difference to DMG
+SGB  | 256          | Only forwards logo to SGB BIOS, performs no checks
+SGB2 | 256          | Same difference to SGB than between MGB and DMG
+CGB0 | 256 + 1792   | Does not init [wave RAM](<#FF30-FF3F - Wave Pattern RAM>)
+CGB  | 256 + 1792   | Split in two parts, with the cartridge header in the middle
+AGB  | 256 + 1792   | Fixes "logo TOCTTOU"
+
+A disassembly of all of them [is available online](https://github.com/ISSOtm/gb-bootroms).
+
+## Monochrome models (DMG0, DMG, MGB)
+
+The monochrome boot ROMs read [the logo from the header](<#0104-0133 - Nintendo Logo>), unpack it into VRAM, and then start slowly scrolling it down.
+Since reads from an absent cartridge usually return $FF, this explains why powering the console on without a cartridge scrolls a black box.
+Additionally, fauly or dirty connections can cause the data read to be corrupted, resulting in a jumbled-up logo.
+
+*Once the logo has finished scrolling*, the boot ROM plays the famous "ba-ding!" sound, and reads the logo **again**, this time comparing it to a copy it stores.
+Then, it also computes the header checksum, and compares it to [the checksum stored in the header](<#014D - Header Checksum>).
+If either of these checks fail, the boot ROM **locks up**, and control is never passed to the cartridge ROM.
+
+Finally, the boot ROM writes to the `BANK` register at $FF50, which unmaps the boot ROM.
+The `ldh [$FF50], a` instruction being located at $00FE (and being two bytes long), [the first instruction executed from the cartridge ROM is at $0100](<#0100-0103 - Entry Point>).
+
+Since the A register is used to write to $FF50, its value is passed to the cartridge ROM; the only difference between the DMG and MGB boot ROMs is that the former writes $01, and the latter uses $FF.
+
+### DMG0
+
+The DMG0 is a rare "early bird" variant of the DMG boot ROM present in few early DMGs.
+The behavior of the boot ROM is globally the same, but significant portions of the code have been rearranged.
+
+Interestingly, the DMG0 boot ROM performs both the logo and checksum checks before displaying anything, and if either fails, the screen is made to blink while the boot ROM locks up, alternating between solid white and solid black.
```

Comment:
*if either fails, the screen is made to blink while the boot ROM locks up* looks like a dependent clause so no comma before the connecting and.